lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.ts

```diff
@@ -51,62 +52,59 @@ import { DATE_FORMAT_CLOUD } from '../../../../models/date-format-cloud.model';
 export class DateCloudWidgetComponent extends WidgetComponent implements OnInit, OnDestroy {
     typeId = 'DateCloudWidgetComponent';
 
-    minDate: Moment;
-    maxDate: Moment;
+    minDate: string;
+    maxDate: string;
 
     private onDestroy$ = new Subject<boolean>();
 
     constructor(public formService: FormService,
-                private dateAdapter: DateAdapter<Moment>,
-                private userPreferencesService: UserPreferencesService) {
+                private dateAdapter: DateAdapter<DateFnsAdapter>,
+                private userPreferencesService: UserPreferencesService,
+                @Inject(MAT_DATE_FORMATS) private dateFormatConfig: MatDateFormats,
+                protected dateFormatTranslationService: DateFormatTranslationService) {
         super(formService);
     }
 
     ngOnInit() {
         this.userPreferencesService
             .select(UserPreferenceValues.Locale)
             .pipe(takeUntil(this.onDestroy$))
-            .subscribe(locale => this.dateAdapter.setLocale(locale));
+            .subscribe(locale => this.dateAdapter.setLocale(DateFnsUtils.getLocaleFromString(locale)));
 
-        const momentDateAdapter = this.dateAdapter as MomentDateAdapter;
-        momentDateAdapter.overrideDisplayFormat = this.field.dateDisplayFormat;
+        this.dateFormatConfig.display.dateInput = this.field.dateDisplayFormat;
 
         if (this.field) {
             if (this.field.dynamicDateRangeSelection) {
-                const today = this.getTodaysFormattedDate();
+                const today = new Date();
                 if (Number.isInteger(this.field.minDateRangeValue)) {
-                    this.minDate = moment(today).subtract(this.field.minDateRangeValue, 'days');
-                    this.field.minValue = this.minDate.format(DATE_FORMAT_CLOUD);
+                    this.minDate = this.dateFormatTranslationService.format(subDays(today, this.field.minDateRangeValue), DATE_FORMAT_CLOUD);
+                    this.field.minValue = this.minDate;
                 }
                 if (Number.isInteger(this.field.maxDateRangeValue)) {
-                    this.maxDate = moment(today).add(this.field.maxDateRangeValue, 'days');
-                    this.field.maxValue = this.maxDate.format(DATE_FORMAT_CLOUD);
+                    this.maxDate = this.dateFormatTranslationService.format(addDays(today, this.field.maxDateRangeValue), DATE_FORMAT_CLOUD);
+                    this.field.maxValue = this.maxDate;
                 }
             } else {
                 if (this.field.minValue) {
-                    this.minDate = moment(this.field.minValue, DATE_FORMAT_CLOUD);
+                    this.minDate = this.dateFormatTranslationService.format(new Date(this.field.minValue), DATE_FORMAT_CLOUD);
```

Comment:
why do you need translation if you already have `new Date()` and `export const DATE_FORMAT_CLOUD = 'yyyy-MM-dd';` ? you can probably use datefns directly?

Reply:
after dome changes now, DATE_FORMAT_CLOUD is 'YYYY-MM-DD' like before.
It also helps to maintain uniformity across all places if we use format or parse methods from DateFnsUtils.